Refactor-migration-guide.md

```diff
@@ -0,0 +1,342 @@
+# Managing the IF refactor
+
+We recently completed a major refactor of the entire IF codebase, moving from an object oriented to functional programmign style.
+
+If you are a IF user or developer, there are some changes you must be aware of resulting from a recent refactor of the IF codebase. This guide will help you to update your processes to integrate your work with the newly refactored IF.
+
+## Running IF
+
+There have been some name changes to the CLI, specifically:
+
+- `impact-engine` --> `if`
```

Comment:
Nit: Since`if` was changed to `ie` this should be adjusted in the guide as well.

Reply:
```suggestion
- `impact-engine` --> `ie`
```